server/src/main/java/org/elasticsearch/index/translog/source/TranslogSource.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.translog.source;
+
+public interface TranslogSource {
```

Comment:
@bleskes We're discussing with @DaveCTurner whether we need this interface and implementations or we could pass source as a string when constructing BufferedChecksumStreamInput and TranslogCorruptedException.
My argument for having this interface is because if you have String you can pass pretty much everything and even two methods that are using local file could pass different values and this might complicate debugging.
@DaveCTurner is generally for strong static typing, but he says that in this case introducing type hierarchy complicates reading the code.
What is your opinion?